### DISCUSSION TITLE
Zend_Acl_Exception: Resource ... not found

### DISCUSSION THREAD

**Discussion by addison74** [OpenMage/magento-lts#2607] [Questions & Answers] [answered]
I am migrating a Magento 1.9.4.5 to OpenMage 20. The store has many users with various roles in the Backend. In the ```exception.log``` file I got several errors like the one below for several resources.

```2022-09-17T17:19:26+00:00 ERR (3): 
Zend_Acl_Exception: Resource 'admin/system/config/moneybookers' not found in /var/www/html/lib/Zend/Acl.php:365
Stack trace:
#0 /var/www/html/lib/Zend/Acl.php(645): Zend_Acl->get('admin/system/co...')
#1 /var/www/html/lib/Zend/Acl.php(523): Zend_Acl->setRule('OP_ADD', 'TYPE_DENY', Array, Array, NULL, NULL)
#2 /var/www/html/app/code/core/Mage/Admin/Model/Resource/Acl.php(138): Zend_Acl->deny('G3', 'admin/system/co...', NULL, NULL)
#3 /var/www/html/app/code/core/Mage/Admin/Model/Resource/Acl.php(76): Mage_Admin_Model_Resource_Acl->loadRules(Object(Mage_Admin_Model_Acl), Array)
#4 /var/www/html/app/code/core/Mage/Admin/Model/Session.php(169): Mage_Admin_Model_Resource_Acl->loadAcl()
#5 /var/www/html/app/code/core/Mage/Admin/Model/Observer.php(69): Mage_Admin_Model_Session->login('addison74', 'a12301230', Object(Mage_Core_Controller_Request_Http))
#6 /var/www/html/app/code/core/Mage/Core/Model/App.php(1401): Mage_Admin_Model_Observer->actionPreDispatchAdmin(Object(Varien_Event_Observer))
#7 /var/www/html/app/code/core/Mage/Core/Model/App.php(1379): Mage_Core_Model_App->_callObserverMethod(Object(Mage_Admin_Model_Observer), 'actionPreDispat...', Object(Varien_Event_Observer), 'auth')
#8 /var/www/html/app/Mage.php(500): Mage_Core_Model_App->dispatchEvent('controller_acti...', Array)
#9 /var/www/html/app/code/core/Mage/Core/Controller/Varien/Action.php(533): Mage::dispatchEvent('controller_acti...', Array)
#10 /var/www/html/app/code/core/Mage/Adminhtml/Controller/Action.php(179): Mage_Core_Controller_Varien_Action->preDispatch()
#11 /var/www/html/app/code/core/Mage/Core/Controller/Varien/Action.php(416): Mage_Adminhtml_Controller_Action->preDispatch()
#12 /var/www/html/app/code/core/Mage/Core/Controller/Varien/Router/Standard.php(261): Mage_Core_Controller_Varien_Action->dispatch('index')
#13 /var/www/html/app/code/core/Mage/Core/Controller/Varien/Front.php(187): Mage_Core_Controller_Varien_Router_Standard->match(Object(Mage_Core_Controller_Request_Http))
#14 /var/www/html/app/code/core/Mage/Core/Model/App.php(370): Mage_Core_Controller_Varien_Front->dispatch()
#15 /var/www/html/app/Mage.php(734): Mage_Core_Model_App->run(Array)
#16 /var/www/html/index.php(82): Mage::run('', 'store')
#17 {main}
```

As you may know the MoneyBookers module, as well as others, were removed from OpenMage and I had to solve this issue that I have not faced until now. Unfortunately, I couldn't find any idea on the Internet looking for the error message. I finally found a solution, I hope it's the right one. I went to ```Backend > System > Permissions > Roles``` and I opened each "Role Name" then I pressed the [Save Role] button. Now OM no longer displays any errors in the exceptions.log file.

In conclusion, if a module is removed from the installation and it had ACLs, all existing "Role Resources" must be updated using this method.

**Comment by sreichel** [comment, marked as answer]
It's related to #2339.

Maybe we should add a script to delete orphan entries?

**Reply by addison74** [reply]
Unfortunately I did not analyze the source of the issue, where the users' ACLs were stored. Most likely they were in the database, but by saving the roles I rewrote what was before. They were practically orphaned entries left over from the removal of many modules from Magento. I can't call it a major issue, but I wanted to make it public so that others can find the post if they search for the error message.
